disk.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+import pyudev
+import psutil
+import subprocess
+
+# of course, we can use sys.argv but just for quick test this more useful
+freeSpaceGB = 10
+fileName = "testFile"
+fileSizeMB = 10
+
+context = pyudev.Context()
+# I have some flash drivers, so can easily test it
+# in case, if we want use it for some not removable devices,
+# we should use device.attributes.asstring('removable') == "0"
+removable = [device for device in context.list_devices(subsystem='block', DEVTYPE='disk') if device.attributes.asstring('removable') == "1"]
+for device in removable:
+    partitions = [device.device_node for device in context.list_devices(subsystem='block', DEVTYPE='partition', parent=device)]
+    print("All removable partitions: {}".format(", ".join(partitions)))
+    print("Mounted removable partitions:")
+    for p in psutil.disk_partitions():
+        if p.device in partitions:
+            obj_Disk = psutil.disk_usage(p.mountpoint)
+            if (obj_Disk.free / (1024.0 ** 3)) >=freeSpaceGB:
+                # we have enough space, go to create file
+                fullFilePath = p.mountpoint + "/" + fileName
+                open(fullFilePath, 'w').close()
+                o = subprocess.check_output(
```

Comment:
> create Z files of size Y, run Z “dd” processes which where each process will ...

Your code runs only 1 process of `dd` at once. You need to run Z processes.
You need to create Z files. You even don't have such variable (parameter) in our program.